### PR TITLE
50 migrate to new sdk

### LINKED ISSUE #50: Migrate java client
Please [read these guidelines](http://ibm.biz/cdt-issue-guide) before opening an issue.

## Bug Description

`java-cloudant` is EOL, migrate to `cloudant-java-sdk`

## Environment details
- Version(s) that are affected by this issue.
    > `<= 0.100.2-kafka-1.0.0`
- Java version including vendor and platform
    > Any
- Apache Kafka versions (including Scala and any vendor or platform info)
    > Any

### PR DESCRIPTION
**Pull request by emlaver**
## Checklist

- [x] Tick to sign-off your agreement to the [Developer Certificate of Origin (DCO) 1.1](../blob/master/DCO1.1.txt)
- [x] Added tests for code changes _or_ test/build only changes
- [x] Updated the change log file (`CHANGES.md`|`CHANGELOG.md`) _or_ test/build only changes
- [x] Completed the PR template below:

## Description
fixes #50
The underlying Cloudant client library is now EOL. This PR refactored the codebase and tests to use the new `cloudant-java-sdk` library. 

## Schema & API Changes
"No change"

## Testing
Updated existing tests to use new library.
No new tests.

### CODE REVIEW

File: src/main/java/com/ibm/cloudant/kafka/connect/CloudantSourceTask.java [hunk 1]
```diff
@@ -166,7 +150,18 @@ public void start(Map<String, String> props) {
...
+                //Password pass = config.getPassword(InterfaceConst.PASSWORD);
+                if (!Objects.requireNonNull(config.getPassword(InterfaceConst.PASSWORD)).value().isEmpty()) {
+                    password = config.getPassword(InterfaceConst.PASSWORD).value();
+                } else {
+                    throw new ConfigException("Password required in config.");
+                }
+                service = JavaCloudantUtil.getClientInstance(url, userName, password);
+            }  else {
+                throw new ConfigException("Username and password required in config.");
+            }
```
Comment (ricellis) [inline src/main/java/com/ibm/cloudant/kafka/connect/CloudantSourceTask.java:164]: Similar to the other validation piece - I think we should drop this to keep the migration "simple" and add it in the "Kafka way" later.

File: src/main/java/com/ibm/cloudant/kafka/connect/CloudantSourceTask.java [hunk 2]
```diff
@@ -211,27 +205,24 @@ public void start(Map<String, String> props) {
 
     @Override
     public void stop() {
+        // TODO needs a custom changes feed
```
Comment (ricellis) [inline src/main/java/com/ibm/cloudant/kafka/connect/CloudantSourceTask.java:208]: We could potentially store the running request in the task and call cancel on it if stop is called.
I don't think `changes().stop()` even went that far in the old client though.

File: src/test/java/com/ibm/cloudant/kafka/connect/utils/CloudantDbUtils.java [hunk 3]
```diff
@@ -13,20 +13,43 @@
...
+	public static void dropDatabase(Map<String, String> props)
+		throws MalformedURLException {
+		Cloudant service = JavaCloudantUtil.getClientInstance(props);
+		String dbName = JavaCloudantUtil.getDbNameFromUrl(props.get(InterfaceConst.URL));
+		DeleteDatabaseOptions deleteDbOptions = new DeleteDatabaseOptions.Builder()
+			.db(dbName)
+			.build();
+		Response<Ok> result = service.deleteDatabase(deleteDbOptions).execute();
+		if (!result.getResult().isOk()) {
+			LOG.error(String.format("Error during deletion of database %s.  Error code: %d Error response: %s",
```
Comment (ricellis) [inline src/test/java/com/ibm/cloudant/kafka/connect/utils/CloudantDbUtils.java:43]: What is the motivation for logging in the test vs just failing?

**Review by ricellis — COMMENTED** [review]
Some initial thoughts. The diff is (unnsurprisingly) quite challenging so maybe you can address these and then I can take another look.

File: src/test/java/com/ibm/cloudant/kafka/connect/utils/CloudantDbUtils.java (hunk 3 above)
Comment (emlaver) [reply src/test/java/com/ibm/cloudant/kafka/connect/utils/CloudantDbUtils.java]: I removed this log and added an assertion in 9720637

**Review by ricellis — COMMENTED** [review]
The copyright dates need a refresh frrom 2021 to 2022 and a couple of minor new questions, but otherwise this is looking OK to swap the client out ready for future improvements.

File: src/main/java/com/ibm/cloudant/kafka/common/utils/JavaCloudantUtil.java [hunk 4]
```diff
@@ -191,17 +184,21 @@ public static String getDbNameFromUrl(String dbUrl) {
...
 		try {
-			result = service.putDatabase(dbOptions).execute();
+			service.putDatabase(dbOptions).execute();
 		} catch (ServiceResponseException sre) {
-			//sre.printStackTrace();
 			// error can happen if db exists
-			LOG.error(String.format("Error during creation of database %s.  Error code: %d Error response: %s",
-				dbName, sre.getStatusCode(), sre.getMessage()));
+			// pass in error message e.g. "Error during creation of database <dbname>"
+			if (sre.getStatusCode() == 412) {
```
[the remark this answers is not in the document]
Comment (emlaver) [reply src/main/java/com/ibm/cloudant/kafka/common/utils/JavaCloudantUtil.java:195]: Yes exactly!

File: CHANGES.md [hunk 5]
```diff
@@ -1,4 +1,6 @@
 # UNRELEASED
+- [NEW] The underlying Cloudant client library is now EOL. Refactored implementation and test suite to
```
Comment (ricellis) [inline CHANGES.md:2]: Nit, it is new, but it's also an upgrade and I think we can keep it short and sweet.
```suggestion
- [UPGRADED] Refactored to use the new `cloudant-java-sdk` library.
```

**Review by ricellis — APPROVED** [review]
+1, a changelog nit sorry I didn't spot it earlier.
I think this will be ok to merge if you are satisified with your testing outcomes.

**Pull request merged**